packages/gatsby-theme-carbon/package.json

```diff
@@ -12,55 +12,56 @@
   },
   "license": "Apache 2.0",
   "peerDependencies": {
-    "gatsby": "^4.21.1",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "gatsby": "^5.9.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
   "devDependencies": {
-    "gatsby": "^4.25.7",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "gatsby": "^5.12.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
   "dependencies": {
     "@babel/core": "^7.23.2",
+    "@babel/plugin-proposal-class-properties": "^7.18.6",
+    "@babel/plugin-syntax-jsx": "^7.22.5",
     "@carbon/elements": "^11.32.1",
     "@carbon/grid": "^11.21.1",
     "@carbon/pictograms-react": "^11.51.1",
     "@carbon/react": "^1.41.1",
     "@carbon/themes": "^11.26.1",
-    "@mdx-js/mdx": "^1.6.22",
-    "@mdx-js/react": "^1.6.22",
+    "@garcia-enterprise/gatsby-plugin-sass-resources": "^4.0.2",
+    "@mdx-js/mdx": "^2.3.0",
+    "@mdx-js/react": "^2.3.0",
     "@reach/router": "^1.3.4",
     "@vimeo/player": "^2.20.1",
-    "beautiful-react-hooks": "^0.23.1",
+    "beautiful-react-hooks": "^4.3.0",
     "classnames": "^2.3.2",
     "copy-to-clipboard": "^3.3.3",
-    "focus-trap-react": "^8.11.3",
-    "gatsby-plugin-catch-links": "^4.25.0",
+    "focus-trap-react": "^10.2.1",
+    "gatsby-plugin-catch-links": "^5.12.0",
     "gatsby-plugin-lunr": "^1.5.2",
-    "gatsby-plugin-manifest": "^4.25.0",
-    "gatsby-plugin-mdx": "3.20.0",
-    "gatsby-plugin-offline": "^5.25.0",
-    "gatsby-plugin-sass": "5.25.0",
-    "gatsby-plugin-sass-resources": "^3.0.1",
-    "gatsby-plugin-sharp": "^4.25.1",
-    "gatsby-remark-copy-linked-files": "^5.25.0",
-    "gatsby-remark-images": "^6.25.0",
-    "gatsby-remark-responsive-iframe": "^5.25.0",
-    "gatsby-remark-smartypants": "^5.25.0",
+    "gatsby-plugin-manifest": "^5.12.0",
+    "gatsby-plugin-mdx": "^5.12.0",
```

Comment:
Hi @eng618 I see you have already updated the version here and still we are using defaultLayouts in the config file. Can you help me with some context around what this version upgrade has brought/deprecated for which we need to make changes?